pep-0596.rst

```diff
@@ -0,0 +1,191 @@
+currently used.  So changing default text file encoding based on current
+code page may cause confusion.
+
+Consistent default text encoding will make Python behavior more expectable
+and easy to learn.
+
+
+Use UTF-8 by default is easier to new programmers
+-------------------------------------------------
+
+Python is one of the most popular first programming language.
+
+New programmers may not know about encoding.  When they download text data
+written in UTF-8 from the internet, they are forced to know encoding.
+
+Popular text editers like VS Code or Atom use UTF-8 by default.
+Even notepad.exe uses UTF-8 by default from Windows 10 2019 may update.
+(Note that Python 3.9 will be released in 2021.)
+
+It would be nice if new programmers are not forced to know about encoding
+until they need to handle text files encoded in encoding other than UTF-8.
+
+
+Specification
+=============
+
+From Python 3.9, default encoding of ``TextIOWrapper`` and ``open()`` is
+changed from ``locale.getpreferredencoding(False)`` to "UTF-8".
+
+When there is device encoding (``os.device_encoding(buffer.fileno())``),
+itstill precedes than this default encoding.
+
+
+Not affected areas
+------------------
+
+Unlike UTF-8 mode, ``locale.getpreferredencoding(False)`` returns
+locale encoding.  Locale encoding is still respected when it makes sense.
+
+``stdin``, ``stdout``, and ``stderr`` keep respecting locale too.  For example,
+these commands don't cause mojibake regardless code page::
+
+   > python -c "print('こんにちは')" | more
+   こんにちは
+   > python -c "print('こんにちは')" > temp.txt
+   > type temp.txt
+   こんにちは
+
+Pipes and TTY should use locale encoding:
+
+* ``subprocess`` and ``os.popen`` use locale encoding because subprocess
+  will use locale encoding.
+* ``getpass.getpass`` use locale encoding when using TTY.
+
+Of course, this PEP is about file content, not file path.  So filenames
+or file paths are not affected.
+
+
+Affected APIs
+--------------
+
+All code using default encoding of ``TextIOWrapper`` or ``open`` are
+affected.  This is incomplete list of APIs affected by this PEP:
+
+* ``lzma.open``, ``gzip.open``, ``bz2.open``, ``ZipFile.read_text``.
+* ``socket.makefile``
+* ``tempfile.TemporaryFile``, ``tempfile.NamedTemporaryFile``
+* ``trace.CoverageResults.write_results_file``
+
+
+Warning
+-------
+
+From 3.8, ``DeprecationWarning`` is shown when encoding is omitted and
+``locale.getpreferredencoding(False)`` is not UTF-8.  This helps not only
+writing forward compatible code, but also investigating unexpected
+``UnicodeDecodeError`` caused by the default encoding.  (See Motivation section
+above.)
+
+
+Backward compatibility
+======================
+
+There may be scripts relying locale or code page which is not UTF-8.
+
+They must be rewritten to use ``encoding=locale.getpreferredencoding(False)``,
+or ``encoding="mbcs"`` on Windows when porting them to Python 3.9.
+
+
+Alternative ideas
+=================
```

Comment:
Can you document somewhere that we don't want UTF-8 BOM? Unicode advices to *not* use BOM if I recall properly.

Reply:
Also, do any Windows tools still add BOM? I know Notepad did at one point, and if that's common, how would it affect interoperability?

Also, can we move these discussions to python-dev or Discourse? It's nearly impossible to follow the points being made and the implications when they are against the PR like this :-(

Reply:
Of course, I will post the article to Discourse when I merge this PR.